cluster/assemblestate/devices.go

```diff
@@ -0,0 +1,214 @@
+package assemblestate
+
+import (
+	"time"
+)
+
+type DeviceTracker struct {
+	responses chan struct{}
+	queries   chan struct{}
```

Comment:
these need comments describing them, especially because they don't actually seem to be sent over what their name suggests but are just flags?